src/Common/src/System/Collections/Generic/ArrayBuilder.cs

```diff
@@ -0,0 +1,158 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Collections.Generic
+{
+    internal struct ArrayBuilder<T>
+    {
+        private const int DefaultCapacity = 4;
```

Comment:
I am not sure whether this is a good default choice in this case. This class is meant to be used for building arrays. For these kind of uses, it is very common that the array ends up containing just one element ... .